### ISSUE TITLE
unable to use new tenant kubeconfig; tls: failed to find any PEM data in certificate input

### ISSUE THREAD

**Issue by abexamir** [clastix/capsule#371]
# Bug description

After installing Capsule into the cluster and creating the first tenant using the getting started guide on README, I created a user using the hack script as mentioned in the guide, but after assigning a new kubeconfig file( `export KUBECONFIG=alice-oil.kubeconfig`) I can't use the cluster and make any request to apiserver. for a simple `kubectl get pods ` command I get this error:
```
E0807 13:33:17.999273  124903 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
E0807 13:33:18.004888  124903 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
E0807 13:33:18.015533  124903 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
E0807 13:33:18.035779  124903 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
E0807 13:33:18.076199  124903 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
E0807 13:33:18.156930  124903 cert_rotation.go:168] key failed with : tls: failed to find any PEM data in certificate input
error: You must be logged in to the server (the server has asked for the client to provide credentials (get deployments.apps capsule-controller-manager))
```

# How to reproduce

Just like the how-to guide on the README file, I used the default 'oil' tenant YAML file and the `hack/create_user.sh` script.
1. Provide the Capsule Tenant YAML definitions
2. Provide all managed Kubernetes resources

# Expected behavior

I can use the cluster using the new user in the new tenant I just created.
# Logs
```

2021-08-07T08:58:42.681Z        INFO    setup   Capsule Version  
2021-08-07T08:58:42.681Z        INFO    setup   Build from: 
2021-08-07T08:58:42.681Z        INFO    setup   Build date: 
2021-08-07T08:58:42.681Z        INFO    setup   Go Version: go1.16.6
2021-08-07T08:58:42.681Z        INFO    setup   Go OS/Arch: linux/amd64
I0807 08:58:43.832774       1 request.go:655] Throttling request took 1.048768036s, request: GET:https://10.43.0.1:443/apis/gitjob.cattle.io/v1?timeout=32s
2021-08-07T08:58:44.197Z        INFO    controller-runtime.metrics      metrics server is starting to listen    {"addr": ":8080"}
2021-08-07T08:58:44.389Z        INFO    controller-runtime.builder      skip registering a mutating webhook, admission.Defaulter interface is not implemented   {"GVK": "capsule.clastix.io/v1alpha1, Kind=Tenant"}
2021-08-07T08:58:44.389Z        INFO    controller-runtime.builder      skip registering a validating webhook, admission.Validator interface is not implemented {"GVK": "capsule.clastix.io/v1alpha1, Kind=Tenant"}
2021-08-07T08:58:44.390Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/convert"}
2021-08-07T08:58:44.390Z        INFO    controller-runtime.builder      conversion webhook enabled      {"object": {"name": ""}}
2021-08-07T08:58:44.391Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/pods"}
2021-08-07T08:58:44.391Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/namespaces"}
2021-08-07T08:58:44.392Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/ingresses"}
2021-08-07T08:58:44.392Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/persistentvolumeclaims"}
2021-08-07T08:58:44.392Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/services"}
2021-08-07T08:58:44.392Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/networkpolicies"}
2021-08-07T08:58:44.393Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/tenants"}
2021-08-07T08:58:44.393Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/namespace-owner-reference"}
2021-08-07T08:58:44.393Z        INFO    controller-runtime.webhook      registering webhook     {"path": "/cordoning"}
2021-08-07T08:58:44.398Z        INFO    setup   starting manager
I0807 08:58:44.399851       1 leaderelection.go:243] attempting to acquire leader lease capsule-system/42c733ea.clastix.capsule.io...
2021-08-07T08:58:44.399Z        INFO    controller-runtime.manager      starting metrics server {"path": "/metrics"}
W0807 08:58:44.480879       1 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W0807 08:58:44.481626       1 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W0807 08:58:44.485877       1 warnings.go:70] networking.k8s.io/v1beta1 Ingress is deprecated in v1.19+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W0807 08:58:44.491922       1 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
2021-08-07T08:58:44.579Z        INFO    controller-runtime.webhook.webhooks     starting webhook server
2021-08-07T08:58:44.580Z        INFO    controller-runtime.certwatcher  Updated current TLS certificate
2021-08-07T08:58:44.581Z        INFO    controller-runtime.webhook      serving webhook server  {"host": "", "port": 9443}
2021-08-07T08:58:44.581Z        INFO    controller-runtime.certwatcher  Starting certificate watcher
I0807 08:59:02.400072       1 leaderelection.go:253] successfully acquired lease capsule-system/42c733ea.clastix.capsule.io
2021-08-07T08:59:02.400Z        DEBUG   controller-runtime.manager.events       Normal  {"object": {"kind":"ConfigMap","namespace":"capsule-system","name":"42c733ea.clastix.capsule.io","uid":"e349440b-e28c-4704-93ef-632904003331","apiVersion":"v1","resourceVersion":"1894696"}, "reason": "LeaderElection", "message": "capsule-controller-manager-6c9cb7567b-njl58_958b9c74-1eaf-4fde-bfde-84e7e7d9bbe2 became leader"}
2021-08-07T08:59:02.400Z        DEBUG   controller-runtime.manager.events       Normal  {"object": {"kind":"Lease","namespace":"capsule-system","name":"42c733ea.clastix.capsule.io","uid":"122b3491-d95b-46e8-9d49-599e829b3f13","apiVersion":"coordination.k8s.io/v1","resourceVersion":"1894697"}, "reason": "LeaderElection", "message": "capsule-controller-manager-6c9cb7567b-njl58_958b9c74-1eaf-4fde-bfde-84e7e7d9bbe2 became leader"}
2021-08-07T08:59:02.400Z        INFO    controller-runtime.manager.controller.secret    Starting EventSource    {"reconciler group": "", "reconciler kind": "Secret", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.400Z        INFO    controller-runtime.manager.controller.secret    Starting EventSource    {"reconciler group": "", "reconciler kind": "Secret", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.401Z        INFO    controller-runtime.manager.controller.service   Starting EventSource    {"reconciler group": "", "reconciler kind": "Service", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.401Z        INFO    controllers.Rbac        setting up ClusterRoles {"ClusterRole": "capsule-namespace-provisioner"}
2021-08-07T08:59:02.401Z        INFO    controller-runtime.manager.controller.endpointslice     Starting EventSource    {"reconciler group": "discovery.k8s.io", "reconciler kind": "EndpointSlice", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.401Z        INFO    controller-runtime.manager.controller.tenant    Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.402Z        INFO    controller-runtime.manager.controller.endpoints Starting EventSource    {"reconciler group": "", "reconciler kind": "Endpoints", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.402Z        INFO    controller-runtime.manager.controller.clusterrole       Starting EventSource    {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRole", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.402Z        INFO    controller-runtime.manager.controller.clusterrolebinding        Starting EventSource    {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRoleBinding", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.402Z        INFO    controller-runtime.manager.controller.capsuleconfiguration      Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "CapsuleConfiguration", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.402Z        INFO    controller-runtime.manager.controller.tenant    Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.403Z        INFO    controller-runtime.manager.controller.tenant    Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.680Z        INFO    controllers.Rbac        setting up ClusterRoles {"ClusterRole": "capsule-namespace-deleter"}
2021-08-07T08:59:02.680Z        INFO    controllers.Rbac        setting up ClusterRoleBindings
W0807 08:59:02.682073       1 warnings.go:70] discovery.k8s.io/v1beta1 EndpointSlice is deprecated in v1.21+, unavailable in v1.25+; use discovery.k8s.io/v1 EndpointSlice
2021-08-07T08:59:02.688Z        INFO    controller-runtime.manager.controller.endpoints Starting Controller     {"reconciler group": "", "reconciler kind": "Endpoints"}
2021-08-07T08:59:02.688Z        INFO    controller-runtime.manager.controller.service   Starting Controller     {"reconciler group": "", "reconciler kind": "Service"}
2021-08-07T08:59:02.689Z        INFO    controller-runtime.manager.controller.clusterrole       Starting Controller     {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRole"}
2021-08-07T08:59:02.782Z        INFO    controller-runtime.manager.controller.clusterrolebinding        Starting EventSource    {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRoleBinding", "source": {}}
W0807 08:59:02.782882       1 warnings.go:70] discovery.k8s.io/v1beta1 EndpointSlice is deprecated in v1.21+, unavailable in v1.25+; use discovery.k8s.io/v1 EndpointSlice
2021-08-07T08:59:02.783Z        INFO    controller-runtime.manager.controller.endpointslice     Starting Controller     {"reconciler group": "discovery.k8s.io", "reconciler kind": "EndpointSlice"}
2021-08-07T08:59:02.881Z        INFO    controller-runtime.manager.controller.tenant    Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "source": "kind source: /, Kind="}
2021-08-07T08:59:02.883Z        INFO    controller-runtime.manager.controller.capsuleconfiguration      Starting Controller     {"reconciler group": "capsule.clastix.io", "reconciler kind": "CapsuleConfiguration"}
2021-08-07T08:59:02.884Z        INFO    controller-runtime.manager.controller.clusterrolebinding        Starting Controller     {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRoleBinding"}
2021-08-07T08:59:02.982Z        INFO    controller-runtime.manager.controller.tenant    Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "source": "kind source: /, Kind="}
2021-08-07T08:59:03.083Z        INFO    controller-runtime.manager.controller.tenant    Starting EventSource    {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "source": "kind source: /, Kind="}
2021-08-07T08:59:03.187Z        INFO    controller-runtime.manager.controller.tenant    Starting Controller     {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant"}
2021-08-07T08:59:03.285Z        INFO    controller-runtime.manager.controller.clusterrolebinding        Starting workers        {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRoleBinding", "worker count": 1}
2021-08-07T08:59:03.288Z        INFO    controller-runtime.manager.controller.tenant    Starting workers        {"reconciler group": "capsule.clastix.io", "reconciler kind": "Tenant", "worker count": 1}
2021-08-07T08:59:03.302Z        INFO    controller-runtime.manager.controller.secret    Starting Controller     {"reconciler group": "", "reconciler kind": "Secret"}
2021-08-07T08:59:03.302Z        INFO    controller-runtime.manager.controller.secret    Starting workers        {"reconciler group": "", "reconciler kind": "Secret", "worker count": 1}
2021-08-07T08:59:03.303Z        INFO    controllers.CA  Reconciling CA Secret   {"Request.Namespace": "capsule-system", "Request.Name": "capsule-ca"}
2021-08-07T08:59:03.380Z        INFO    controllers.CA  Handling CA Secret      {"Request.Namespace": "capsule-system", "Request.Name": "capsule-ca"}
2021-08-07T08:59:03.380Z        INFO    controllers.CA  Updating CA secret with new PEM and RSA {"Request.Namespace": "capsule-system", "Request.Name": "capsule-ca"}
2021-08-07T08:59:03.380Z        INFO    controller-runtime.manager.controller.secret    Starting Controller     {"reconciler group": "", "reconciler kind": "Secret"}
2021-08-07T08:59:03.380Z        INFO    controller-runtime.manager.controller.secret    Starting workers        {"reconciler group": "", "reconciler kind": "Secret", "worker count": 1}
2021-08-07T08:59:03.380Z        INFO    controllers.Tls Reconciling TLS Secret  {"Request.Namespace": "capsule-system", "Request.Name": "capsule-tls"}
2021-08-07T08:59:03.382Z        INFO    controller-runtime.manager.controller.service   Starting workers        {"reconciler group": "", "reconciler kind": "Service", "worker count": 1}
2021-08-07T08:59:03.382Z        INFO    controller-runtime.manager.controller.clusterrole       Starting workers        {"reconciler group": "rbac.authorization.k8s.io", "reconciler kind": "ClusterRole", "worker count": 1}
2021-08-07T08:59:03.382Z        INFO    controller-runtime.manager.controller.endpoints Starting workers        {"reconciler group": "", "reconciler kind": "Endpoints", "worker count": 1}
2021-08-07T08:59:03.383Z        INFO    controller-runtime.manager.controller.capsuleconfiguration      Starting workers        {"reconciler group": "capsule.clastix.io", "reconciler kind": "CapsuleConfiguration", "worker count": 1}
2021-08-07T08:59:03.383Z        INFO    controllers.CapsuleConfiguration        CapsuleConfiguration reconciliation started     {"request.name": "capsule-default"}
2021-08-07T08:59:03.384Z        INFO    controllers.CapsuleConfiguration        CapsuleConfiguration reconciliation finished    {"request.name": "capsule-default"}
2021-08-07T08:59:03.384Z        INFO    controller-runtime.manager.controller.endpointslice     Starting workers        {"reconciler group": "discovery.k8s.io", "reconciler kind": "EndpointSlice", "worker count": 1}
2021-08-07T08:59:03.385Z        INFO    controllers.Tls Reconciliation completed, processing back in 4319h59m6.618445514s       {"Request.Namespace": "capsule-system", "Request.Name": "capsule-tls"}
2021-08-07T08:59:05.498Z        INFO    controllers.CA  Reconciliation completed, processing back in 87647h57m45s       {"Request.Namespace": "capsule-system", "Request.Name": "capsule-ca"}
2021-08-07T08:59:20.789Z        DEBUG   controller-runtime.webhook.webhooks     received request        {"webhook": "/tenants", "UID": "ec8ad4f0-5eeb-465a-8d41-85f1853b270f", "kind": "capsule.clastix.io/v1beta1, Kind=Tenant", "resource": {"group":"capsule.clastix.io","version":"v1beta1","resource":"tenants"}}
2021-08-07T08:59:20.798Z        DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/tenants", "code": 200, "reason": "", "UID": "ec8ad4f0-5eeb-465a-8d41-85f1853b270f", "allowed": true}
2021-08-07T08:59:21.881Z        INFO    controllers.Tenant      Ensuring all Namespaces are collected   {"Request.Name": "oil"}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Starting processing of Namespaces       {"Request.Name": "oil", "items": 0}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Starting processing of Network Policies {"Request.Name": "oil", "items": 1}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Starting processing of Limit Ranges     {"Request.Name": "oil", "items": 1}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Starting processing of Resource Quotas  {"Request.Name": "oil", "items": 3}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Ensuring additional RoleBindings for owner      {"Request.Name": "oil"}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Ensuring RoleBinding for owner  {"Request.Name": "oil"}
2021-08-07T08:59:22.043Z        INFO    controllers.Tenant      Ensuring Namespace count        {"Request.Name": "oil"}
2021-08-07T08:59:22.165Z        INFO    controllers.Tenant      Tenant reconciling completed    {"Request.Name": "oil"}
2021-08-07T08:59:22.204Z        INFO    controllers.Tenant      Ensuring all Namespaces are collected   {"Request.Name": "oil"}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Starting processing of Namespaces       {"Request.Name": "oil", "items": 0}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Starting processing of Network Policies {"Request.Name": "oil", "items": 1}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Starting processing of Limit Ranges     {"Request.Name": "oil", "items": 1}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Starting processing of Resource Quotas  {"Request.Name": "oil", "items": 3}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Ensuring additional RoleBindings for owner      {"Request.Name": "oil"}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Ensuring RoleBinding for owner  {"Request.Name": "oil"}
2021-08-07T08:59:22.253Z        INFO    controllers.Tenant      Ensuring Namespace count        {"Request.Name": "oil"}
2021-08-07T08:59:22.308Z        INFO    controllers.Tenant      Tenant reconciling completed    {"Request.Name": "oil"}


```
# Additional context

- Kubernetes version: 1.21
- Rancher RKE Cluster

**Comment by prometherion** [comment]
Hey @abexamir, thanks for reporting this issue!

May I ask you to share the generated `kubeconfig` and ensuring the content stored on keys `client-certificate` and `client-key` are ok and real certificate and key?

**Comment by ptx96** [comment]
Hi @abexamir and thank you for your question!

Can you please verify with

`kubectl get csr <USER>-<TENANT> -o json`

that your ".spec.certificate" is not empty?

**Comment by bsctl** [comment]
@abexamir we think it should be fixed with #388. Could you please check? Thanks
